### ISSUE TITLE
@aws-sdk/client-sesv2 - Error: Cannot find module '@aws-sdk/util-endpoints'

### ISSUE THREAD

**Issue by subutuo** [aws/aws-sdk-js-v3#4053]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

After upgrading @aws-sdk/client-sesv2  to version 3.191.0
Got a message `Cannot find module '@aws-sdk/util-endpoints'`

[@aws-sdk/client-sesv2@3.191.0 history link](https://github.com/aws/aws-sdk-js-v3/commit/976549dbbf654a2978fbfd5896c6797b4824e229#diff-3e19abc16164d82b22e263820dac71ce2322784d2c343f62eab419b9415a5ee8)

### SDK version number

@aws-sdk/client-sesv2@3.191.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

node.js 14.16.0

### Reproduction Steps

```json
  "dependencies": {
    "@aws-sdk/client-api-gateway": "^3.13.1",
    "@aws-sdk/client-sesv2": "^3.11.0",
  }
```

### Observed Behavior

```
Error: Cannot find module '@aws-sdk/util-endpoints'
Require stack:
- /node_modules/@aws-sdk/client-sesv2/dist-cjs/endpoint/endpointResolver.js
- /node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.shared.js
- /node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.js
- /node_modules/@aws-sdk/client-sesv2/dist-cjs/SESv2Client.js
- /node_modules/@aws-sdk/client-sesv2/dist-cjs/SESv2.js
- /node_modules/@aws-sdk/client-sesv2/dist-cjs/index.js
- /dist/aws/aws.ses.service.js
- /dist/email/email.service.js
- /dist/email/email.module.js
- /dist/auth/auth.module.js
- /dist/app.module.js
- /dist/main.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/node_modules/@aws-sdk/client-sesv2/dist-cjs/endpoint/endpointResolver.js:4:26)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/node_modules/@aws-sdk/client-sesv2/dist-cjs/endpoint/endpointResolver.js',
    '/node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.shared.js',
    '/node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.js',
    '/node_modules/@aws-sdk/client-sesv2/dist-cjs/SESv2Client.js',
    '/node_modules/@aws-sdk/client-sesv2/dist-cjs/SESv2.js',
    '/node_modules/@aws-sdk/client-sesv2/dist-cjs/index.js',
    '/dist/aws/aws.ses.service.js',
    '/dist/email/email.service.js',
    '/dist/email/email.module.js',
    '/dist/auth/auth.module.js',
    '/dist/app.module.js',
    '/dist/main.js'
  ]
}
```

### Expected Behavior

No errors at run time.

**Comment by nathan-fiscaletti** [comment]
Same issue with `@aws-sdk/client-s3` in v3.192.0

**Comment by johncantrell97** [comment]
> Same issue with `@aws-sdk/client-s3` in v3.192.0

Same here but on v3.190.0

**Comment by vishald2509** [comment]
I got the same error. 
Version- 3.192.0

```
   "Runtime.ImportModuleError: Error: Cannot find module '@aws-sdk/util-endpoints'",
    "Require stack:",
    "- /opt/nodejs/node_modules/@aws-sdk/client-s3/dist-cjs/endpoint/endpointResolver.js",
    "- /opt/nodejs/node_modules/@aws-sdk/client-s3/dist-cjs/runtimeConfig.shared.js",
    "- /opt/nodejs/node_modules/@aws-sdk/client-s3/dist-cjs/runtimeConfig.js",
    "- /opt/nodejs/node_modules/@aws-sdk/client-s3/dist-cjs/S3Client.js",
    "- /opt/nodejs/node_modules/@aws-sdk/client-s3/dist-cjs/S3.js",
    "- /opt/nodejs/node_modules/@aws-sdk/client-s3/dist-cjs/index.js",
    "- /var/task/config/s3Client.js",
    "- /var/task/index.js",
    "- /var/runtime/index.mjs",
    "    at _loadUserApp (file:///var/runtime/index.mjs:951:17)",
    "    at async Object.UserFunction.js.module.exports.load (file:///var/runtime/index.mjs:976:21)",
    "    at async start (file:///var/runtime/index.mjs:1137:23)",
    "    at async file:///var/runtime/index.mjs:1143:1"
```

**Comment by ddolcimascolo** [comment]
Hi guys, same here. Should you need anything to help diagnose, just ask. 

Keep up the good work! 

Cheers,
David

**Comment by trivikr** [comment]
Acknowledged that this is a bug.

It can be temporarily fixed by adding `@aws-sdk/util-endpoints` in your project dependency:
* `npm install @aws-sdk/util-endpoints`
* `yarn add @aws-sdk/util-endpoints`
* `pnpm add @aws-sdk/util-endpoints`

We'll attempt to push fix in `v3.193.0` on Thu, Oct 20

**Comment by Ningaro** [comment]
In our project we use `@aws-sdk/client-s3` and we recently had the same error.

Pin dependencie version to `3.188.0` helped us
```diff
"dependencies": {
+    "@aws-sdk/client-s3": "3.188.0"
-    "@aws-sdk/client-s3": "^3.188.0"
}
```

**Comment by trivikr** [comment]
This happens as `resolveEndpoint` is called from `@aws-sdk/util-endpoints`
https://github.com/aws/aws-sdk-js-v3/blob/87fc2979519c9134e7b11528169302882be2214e/clients/client-sesv2/src/endpoint/endpointResolver.ts#L3

However, it's not added as a dependency in package.json
https://github.com/aws/aws-sdk-js-v3/blob/87fc2979519c9134e7b11528169302882be2214e/clients/client-sesv2/package.json#L19-L55

The fix in the SDK would be similar to what's done by @JagWireZ in https://github.com/aws/aws-sdk-js-v3/pull/4055, but it would be codegen.

**Comment by trivikr** [comment]
The issue is in EndpointsV2Generator.java which adds `@aws-sdk/util-endpoints` dependency in the source code, but not in package.json

https://github.com/awslabs/smithy-typescript/blob/334a06d06327b98a9f1f8892205d98d88651b887/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/EndpointsV2Generator.java#L146-L147

**Comment by trivikr** [comment]
Fixes are ready:
* smithy-ts https://github.com/awslabs/smithy-typescript/pull/619
* aws-sdk-js-v3: https://github.com/aws/aws-sdk-js-v3/pull/4058

**Comment by trivikr** [comment]
This issue is fixed in `v3.193.0`

<details>
<summary>Source code</summary>

```mjs
import { SESv2 } from "@aws-sdk/client-sesv2";

const client = new SESv2({});
console.log(await client.listConfigurationSets({}));
```

</details>

<details>
<summary>Cannot find module error with @aws-sdk/client-sesv2@3.192.0</summary>

```console
node:internal/modules/cjs/loader:985
  const err = new Error(message);
              ^

Error: Cannot find module '@aws-sdk/util-endpoints'
Require stack:
- /Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/endpoint/endpointResolver.js
- /Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.shared.js
- /Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.js
- /Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/SESv2Client.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
    at Function.Module._load (node:internal/modules/cjs/loader:833:27)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/endpoint/endpointResolver.js:4:26)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
    at Module.load (node:internal/modules/cjs/loader:1033:32)
    at Function.Module._load (node:internal/modules/cjs/loader:868:12)
    at Module.require (node:internal/modules/cjs/loader:1057:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/endpoint/endpointResolver.js',
    '/Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.shared.js',
    '/Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/runtimeConfig.js',
    '/Users/trivikr/workspace/test/node_modules/@aws-sdk/client-sesv2/dist-cjs/SESv2Client.js'
  ]
}
```

</details>

<details>
<summary>No error with @aws-sdk/client-sesv2@3.193.0</summary>

```console
{
  '$metadata': {
    // Redacted
  },
  ConfigurationSets: []
}
```

</details>
